packages/fluentui/react-northstar/src/components/Alert/Alert.tsx

```diff
@@ -110,157 +127,211 @@ export const alertSlotClassNames: AlertSlotClassNames = {
   body: `${alertClassName}__body`,
 };
 
-class Alert extends AutoControlledComponent<WithAsProp<AlertProps>, AlertState> {
-  static displayName = 'Alert';
-  static deprecated_className = alertClassName;
-
-  static slotClassNames = alertSlotClassNames;
-
-  static propTypes = {
-    ...commonPropTypes.createCommon({ content: 'shorthand' }),
-    actions: PropTypes.oneOfType([customPropTypes.itemShorthand, customPropTypes.collectionShorthand]),
-    icon: customPropTypes.shorthandAllowingChildren,
-    header: customPropTypes.itemShorthand,
-    attached: PropTypes.oneOfType([PropTypes.bool, PropTypes.oneOf(['top', 'bottom'])]),
-    fitted: PropTypes.bool,
-    danger: PropTypes.bool,
-    defaultVisible: PropTypes.bool,
-    dismissible: PropTypes.bool,
-    dismissAction: customPropTypes.itemShorthand,
-    info: PropTypes.bool,
-    onVisibleChange: PropTypes.func,
-    onFocus: PropTypes.func,
-    success: PropTypes.bool,
-    visible: PropTypes.bool,
-    warning: PropTypes.bool,
-    body: customPropTypes.itemShorthand,
-  };
-
-  static defaultProps = {
-    accessibility: alertBehavior,
-    dismissAction: {},
-    body: {},
-  };
+export const Alert: React.FC<WithAsProp<AlertProps>> &
+  FluentComponentStaticProps<AlertProps> & {
+    autoControlledProps: string[];
+    DismissAction: ShorthandValue<typeof AlertDismissAction>;
+  } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Alert.displayName, context.telemetry);
+  setStart();
+  const {
+    warning,
+    danger,
+    info,
+    success,
+    attached,
+    fitted,
+    dismissible,
+    variables,
+    className,
+    design,
+    styles,
+    children,
+    defaultVisible,
+    actions,
+    dismissAction,
+    content,
+    icon,
+    header,
+    body,
+  } = props;
+
+  const [visible, setVisible] = useAutoControlled({
+    defaultValue: defaultVisible,
+    value: props.visible,
+    initialValue: true,
+  });
 
-  static DismissAction = AlertDismissAction;
+  const [bodyId] = React.useState(_.uniqueId('alert-body-'));
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(Alert.handledProps, props);
 
-  static autoControlledProps = ['visible'];
+  const getA11Props = useAccessibility(props.accessibility, {
+    debugName: Alert.displayName,
+    mapPropsToBehavior: () => ({
+      warning,
+      danger,
+      bodyId,
+      visible,
+    }),
+    rtl: context.rtl,
+  });
 
-  getInitialAutoControlledState(): AlertState {
-    return {
-      visible: true,
-      bodyId: _.uniqueId('alert-body-'),
-    };
-  }
+  const { classes, styles: resolvedStyles } = useStyles<AlertStylesProps>(Alert.displayName, {
+    className: alertClassName,
+    mapPropsToStyles: () => ({
+      warning,
+      danger,
+      info,
+      success,
+      attached,
+      fitted,
+      dismissible,
+      actionable: false,
```

Comment:
Is it international to be always `false`?

Reply:
fixed